src/tetris/brick/brick.ts

```diff
@@ -47,29 +47,40 @@ export default class Brick {
 		this._tryToMove(new Vector2(0, 1), true);
 	}
 
-	public addBlock(sprite: Phaser.GameObjects.Sprite, relativePosition: Vector2): void {
-		const block = new Block(sprite, relativePosition);
-		this._blocks = this.blocks.concat(block);
+	public addBlock(sprite: Phaser.GameObjects.Sprite, positions: Vector2[]): void {
+		const block = new Block(sprite, positions.map(p => p.add(this._position)), this);
+		this.blocks.push(block);
 	}
 
 	public isStuck(): boolean {
 		return this._stuck;
 	}
 
+	public checkIfStuck(): boolean {
```

Comment:
What's the difference to `isStuck()`?
Why call tryToMove (and in some cases really move the brick) for a simple check?

Reply:
isStuck itself does not have any side effects, it's a getter. CheckIfStuck checks whether the brick is stuck right now without the flag being already set - a condition which only happens when spawning new bricks. Usually, the brick is stuck after being dropped (by gravity) and the move action (dropping down) being impossible -> in that case, the stuck flag is already set.

Reply:
I see. But wouldn't it make sense to handle this case implicit? Brick spawns and in the next frame gravity tries to move it down => impossible and flag gets set => `isStuck()` is true.
Of course it would take one frame longer to detect a stuck situation but does this delay of only one frame really matter?